src/api/localResolvers/experiment.js

```diff
@@ -25,24 +26,40 @@ export default () => {
 					const experimentSubset = _pick(experiment, ['name', 'distribution', 'variants', 'control']);
 					// get the hash for our current experiment setting
 					const settingHash = hashCode(JSON.stringify(experimentSubset));
+					const population = _get(experiment, 'population') || 1;
 
 					// Add hash to exisitng cookie exps if it's missing
 					if (typeof currentAssignment.hash === 'undefined') {
 						currentAssignment.hash = settingHash;
 					}
 
+					// Add population to existing cookie experiments if it's missing
+					if (typeof currentAssignment.population === 'undefined') {
+						currentAssignment.population = population;
+					}
+
 					// assign an experiment version if:
 					// - we have an experiment
 					// - and the experiment is enabled
 					// - and currentAssignment.version is undefined or the hashes don't match
+					// - current version is 'unassigned' and the population value has changed
 					if (experiment !== null
 						&& experiment.enabled
-						&& (_isUndefined(currentAssignment.version) || settingHash !== currentAssignment.hash)) {
+						&& (
+							_isUndefined(currentAssignment.version)
+							|| settingHash !== currentAssignment.hash
+							|| (
+								population !== currentAssignment.population
+								&& currentAssignment.version === 'unassigned'
+							)
+						)
+					) {
 						// assign the version using the experiment data (undefined if experiment disabled)
 						currentAssignment = {
 							id,
-							version: assignVersion(experiment || {}),
+							version: assignVersion(experiment),
```

Comment:
I think we need to ensure an object is passed here if the experiment value is null. Otherwise we get breakage when trying to pull items off the object inside `assignVersion`

Reply:
Since the if statement checks `experiment !== null && experiment.enabled`, there's no way to get a null/undefined experiment value in here, and it was coming up in code coverage as an uncovered branch.

Reply:
👍